src/gameplayv2/resistance/entities/Entity.ts

```diff
@@ -0,0 +1,37 @@
+class Entity{
+    static nextId = 0;
+    protected components: any[];
+    public role: string;
+    public alliance: string;
+    public description: string;
+
+    constructor() {
+        this.components = [];
+        this.role;
+        this.alliance;
+        this.description;
+    }
+
+    public addComponent(component: any): void {
+        this.components.push(component);
+    }
+
+    public addComponents(...components: any[]): void {
+        components.forEach((component) => this.addComponent(component));
+    }
+      
+    public removeComponent(component: any): void {
+        const componentIndex = this.components.findIndex((c) => c.constructor.name === component.constructor.name);
+        if (componentIndex !== -1) {
+            this.components.splice(componentIndex, 1);
+        }
+    }
```

Comment:
Let's use `this.components.filter()` here and make all the component names an enum.